frame/support/procedural/src/pallet/expand/config.rs

```diff
@@ -41,5 +44,22 @@ pub fn expand_config(def: &mut Def) -> proc_macro2::TokenStream {
 		));
 	}
 
-	Default::default()
+	// we only emit `DefaultConfig` if there are trait items, so an empty `DefaultConfig` is
+	// impossible consequently
+	if config.default_sub_trait.len() > 0 {
+		let trait_items = &config.default_sub_trait;
+		quote!(
+			/// Based on [`Config`]. Auto-generated by
```

Comment:
I would make it clearer that: 

* This is always a *subset* of `Config`.
* It can only contain items that don't rely on externalities such as `frame_system::Config`. 
* Such items should be marked with `#[pallet::no_default]`. Consequently:
	* downstream users MUST specify these 
	* you won't need to implement it in any `impl` of `DefaultConfig`. 

Although, perhaps these must go to the doc-stub of `pallet::default_config` as they are quite detailed. If so, I link from here to the macro-sub as "read more about what this is in <xxx>".